File: php/features.php
<?php
declare(strict_types=1);

// BleachPoems SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BleachPoemsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BleachPoemsBaseFeature();
            case "test":
                return new BleachPoemsTestFeature();
            default:
                return new BleachPoemsBaseFeature();
        }
    }
}

File: py/bleachpoems_sdk/features.py
# BleachPoems SDK feature factory

from bleachpoems_sdk.feature.base_feature import BleachPoemsBaseFeature
from bleachpoems_sdk.feature.test_feature import BleachPoemsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BleachPoemsBaseFeature(),
        "test": lambda: BleachPoemsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
